### PR TITLE
query by createdBy

### PR DESCRIPTION
**Pull request by bingzhang**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by sandeep-ps** [comment]
Hi, @bingzhang, I'm getting an empty list when I use the new query filter.

Please see the POST and GET commands with sample data below:

```
curl --location --request POST 'http://localhost:5000/events' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer <ID Token>' \
--data-raw '{
    "sourceEventId": "33345058",
    "createdBy": "Moises_Kertzmann@yahoo.com",
    "category": "Athletics",
    "sponsor": "Hoppe, Zieme and Thiel",
    "title": "Sleek Frozen Chair",
    "startDate": "2021/01/14T17:00:00",
    "endDate": "2021/01/15T19:00:00",
    "longDescription": "<p>TBA</p>",
    "location": {
        "latitude": 55.7722,
        "longitude": 113.3782,
        "room": "",
        "description": "55072 Daugherty Brook"
    },
    "isVirtual": false,
    "displayOnlyWithSuperEvent": false
}'
```

```
curl --location --request GET 'http://localhost:5000/events?createdBy=Moises_Kertzmann@yahoo.com' \
--header 'ROKWIRE-API-KEY: <API KEY>'
```

Please fix this. Thanks.

**Comment by bingzhang** [comment]
> Hi, @bingzhang, I'm getting an empty list when I use the new query filter.
> 
> Please see the POST and GET commands with sample data below:
> 
> ```
> curl --location --request POST 'http://localhost:5000/events' \
> --header 'Content-Type: application/json' \
> --header 'Authorization: Bearer <ID Token>' \
> --data-raw '{
>     "sourceEventId": "33345058",
>     "createdBy": "Moises_Kertzmann@yahoo.com",
>     "category": "Athletics",
>     "sponsor": "Hoppe, Zieme and Thiel",
>     "title": "Sleek Frozen Chair",
>     "startDate": "2021/01/14T17:00:00",
>     "endDate": "2021/01/15T19:00:00",
>     "longDescription": "<p>TBA</p>",
>     "location": {
>         "latitude": 55.7722,
>         "longitude": 113.3782,
>         "room": "",
>         "description": "55072 Daugherty Brook"
>     },
>     "isVirtual": false,
>     "displayOnlyWithSuperEvent": false
> }'
> ```
> 
> ```
> curl --location --request GET 'http://localhost:5000/events?createdBy=Moises_Kertzmann@yahoo.com' \
> --header 'ROKWIRE-API-KEY: <API KEY>'
> ```
> 
> Please fix this. Thanks.

I can get it.
Screen Shot 2022-03-11 at 11 44 23 AM

File: eventservice/api/utils/query_params.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ def format_query(args, query, include_private_events=False, group_ids=None, all_
         query_parts.append({'_id': ObjectId(super_event_id)})
         query_parts.append({'isSuperEvent': True})
     if args.getlist('createdBy'):
-        query_parts.append({'createdBy': args.getlist('createdBy')})
+        # query_parts.append({'createdBy': args.get('createdBy')})
```
Comment (sandeep-ps) [inline eventservice/api/utils/query_params.py:32]: I think the commented-out code can be removed. Tjhanks.

File: eventservice/api/events_rest_service.py [hunk 2]
```diff
@@ -40,4 +40,4 @@
             resolver_error=501)
 
 if __name__ == '__main__':
-    app.run(port=5000, host=None, server='flask', debug=debug)
+    app.run(port=6000, host=None, server='flask', debug=debug)
```
Comment (sandeep-ps) [inline eventservice/api/events_rest_service.py:43]: Please revert this to port 5000. Thanks.

**Comment by sandeep-ps** [comment]
@bingzhang, Please also update the YAML file with the new query parameter. Thanks.

**Comment by sandeep-ps** [comment]
I tested this with different cases and it looks fine. @bingzhang, please address the remaining comments and I think we can merge this soon.

@minump, do you have additional comments?

**Comment by minump** [comment]
> I tested this with different cases and it's working fine. @bingzhang, please address the remaining comments and I think we can merge this soon.
> 
> @minump, do you have additional comments?



> I tested this with different cases and it's working fine. @bingzhang, please address the remaining comments and I think we can merge this soon.
> 
> @minump, do you have additional comments?

Tested and working fine.

**Review by minump — APPROVED** [review]
Tested and approved.

**Review by sandeep-ps — APPROVED** [review]

**Pull request merged**